core/src/layers/metrics.rs

```diff
@@ -70,148 +67,112 @@ use crate::*;
 /// ```
 #[derive(Clone, Debug, Default)]
-pub struct MetricsLayer {
-    path_label_level: usize,
-}
-
-impl MetricsLayer {
-    /// Set the level of path label.
-    ///
-    /// - level = 0: we will ignore the path label.
-    /// - level > 0: the path label will be the path split by "/" and get the last n level,
-    ///   if n=1 and input path is "abc/def/ghi", and then we will get "abc/" as the path label.
-    pub fn path_label(mut self, level: usize) -> Self {
-        self.path_label_level = level;
-        self
-    }
-}
+pub struct MetricsLayer {}
 
 impl<A: Access> Layer<A> for MetricsLayer {
     type LayeredAccess = observe::MetricsAccessor<A, MetricsInterceptor>;
 
     fn layer(&self, inner: A) -> Self::LayeredAccess {
-        let interceptor = MetricsInterceptor {
-            path_label_level: self.path_label_level,
-        };
+        let interceptor = MetricsInterceptor {};
         observe::MetricsLayer::new(interceptor).layer(inner)
     }
 }
 
 #[derive(Clone, Debug)]
-pub struct MetricsInterceptor {
-    path_label_level: usize,
-}
+pub struct MetricsInterceptor {}
 
 impl observe::MetricsIntercept for MetricsInterceptor {
-    fn observe_operation_duration_seconds(
-        &self,
-        info: Arc<AccessorInfo>,
-        path: &str,
-        op: Operation,
-        duration: Duration,
-    ) {
-        let labels = OperationLabels {
-            info,
-            path: observe::path_label_value(path, self.path_label_level),
-            operation: op,
-            error: None,
+    fn observe(&self, labels: observe::MetricLabels, value: observe::MetricValue) {
+        let labels = OperationLabels(labels).into_labels();
+
+        match value {
+            observe::MetricValue::OperationBytes(v) => {
+                histogram!(value.name(), labels).record(v as f64)
+            }
+            observe::MetricValue::OperationBytesRate(v) => {
+                histogram!(value.name(), labels).record(v)
+            }
+            observe::MetricValue::OperationEntries(v) => {
+                histogram!(value.name(), labels).record(v as f64)
+            }
+            observe::MetricValue::OperationEntriesRate(v) => {
+                histogram!(value.name(), labels).record(v)
+            }
+            observe::MetricValue::OperationDurationSeconds(v) => {
+                histogram!(value.name(), labels).record(v)
+            }
+            observe::MetricValue::OperationErrorsTotal => {
+                counter!(value.name(), labels).increment(1)
+            }
+            observe::MetricValue::OperationExecuting(v) => {
+                if v.is_positive() {
+                    gauge!(value.name(), labels).increment(v as f64)
+                } else {
+                    gauge!(value.name(), labels).decrement(v as f64)
+                }
+            }
+            observe::MetricValue::OperationTtfbSeconds(v) => {
+                histogram!(value.name(), labels).record(v)
```

Comment:
We have some good default histogram buckets setting, how could we use them in `metrics-rs`?

Reply:
Due to the design of the metrics crate, we cannot set the corresponding buckets at the metrics crate level. We can only set the metric buckets in the specific exporter implementation, like https://docs.rs/metrics-exporter-prometheus/latest/metrics_exporter_prometheus/struct.PrometheusBuilder.html#method.set_buckets_for_metric

Reply:
Oh, it's so ...

---

Ok, would you like to add some docs to point our users to use `observe::DEFAULT_BYTES_RATE_BUCKETS` as the defaull buckets?